app/controllers/collection_participants_controller.rb

```diff
@@ -14,37 +14,38 @@ def owners_required
     redirect_to collection_participants_path(@collection)
     false
   end
-  
+
   def no_participant
     flash[:error] = t('no_participant', :default => "Which participant did you want to work with?")
     redirect_to root_path
   end
-  
+
   def load_participant_and_collection
     if params[:collection_participant]
-      @participant = CollectionParticipant.find(params[:collection_participant][:id])
+      @participant = CollectionParticipant.where(id: params[:collection_participant][:id]).first
```

Comment:
Ideally, we prefer not to make bug fixes in PRs whose original purpose is to add tests, as bug fixes require manual testing for sign-off so there's more overhead involved in getting them into production, whereas test-only PRs can be merged straight away. Something to bear in mind for next time, no need to change this now!

I agree it's best to let the code fall through to `no_participant` since it's there, and thank you for spotting the 500 error. These lines of code were already covered, probably by a Cucumber test, and the tests have passed so I think we can be confident that you haven't broken anything as well as making things more robust. 🎉